x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/post/post_attack_discovery_bulk.ts

```diff
@@ -0,0 +1,129 @@
+  API_VERSIONS,
+  ATTACK_DISCOVERY_ALERTS_ENABLED_FEATURE_FLAG,
+  ATTACK_DISCOVERY_BULK,
+  PostAttackDiscoveryBulkRequestBody,
+  PostAttackDiscoveryBulkResponse,
+} from '@kbn/elastic-assistant-common';
+import { buildRouteValidationWithZod } from '@kbn/zod-helpers';
+
+import { performChecks } from '../../helpers';
+import { buildResponse } from '../../../lib/build_response';
+import { ElasticAssistantRequestHandlerContext } from '../../../types';
+
+export const postAttackDiscoveryBulkRoute = (
+  router: IRouter<ElasticAssistantRequestHandlerContext>
+): void => {
+  router.versioned
+    .post({
+      access: 'internal',
+      path: ATTACK_DISCOVERY_BULK,
+      security: {
+        authz: {
+          requiredPrivileges: ['elasticAssistant'],
+        },
+      },
+    })
+    .addVersion(
+      {
+        version: API_VERSIONS.internal.v1,
+        validate: {
+          request: {
+            body: buildRouteValidationWithZod(PostAttackDiscoveryBulkRequestBody),
+          },
+          response: {
+            200: {
+              body: {
+                custom: buildRouteValidationWithZod(PostAttackDiscoveryBulkResponse),
+              },
+            },
+          },
+        },
+      },
+      async (
+        context,
+        request,
+        response
+      ): Promise<IKibanaResponse<PostAttackDiscoveryBulkResponse>> => {
+        const ctx = await context.resolve(['core', 'elasticAssistant', 'licensing']);
+        const resp = buildResponse(response);
+        const assistantContext = await context.elasticAssistant;
+        const { featureFlags } = await context.core;
+        const logger: Logger = assistantContext.logger;
+
+        // Perform license and authenticated user checks:
+        const checkResponse = await performChecks({
+          context: ctx,
+          request,
+          response,
+        });
+
+        if (!checkResponse.isSuccess) {
+          return checkResponse.response;
+        }
+
+        const attackDiscoveryAlertsEnabled = await featureFlags.getBooleanValue(
+          ATTACK_DISCOVERY_ALERTS_ENABLED_FEATURE_FLAG,
+          false
+        );
+
+        if (!attackDiscoveryAlertsEnabled) {
+          return resp.error({
+            body: `Attack discovery alerts feature is disabled`,
+            statusCode: 403,
+          });
+        }
+
+        try {
+          const currentUser = await checkResponse.currentUser;
+          const dataClient = await assistantContext.getAttackDiscoveryDataClient();
+
+          if (!dataClient) {
+            return resp.error({
+              body: `Attack discovery data client not initialized`,
+              statusCode: 500,
+            });
+          }
+
+          const kibanaAlertWorkflowStatus = request.body.update?.kibana_alert_workflow_status;
+          const visibility = request.body.update?.visibility;
+          const ids = request.body.update?.ids;
+
+          if (ids == null || ids.length === 0) {
+            return response.ok({
```

Comment:
should this be an error?

Reply:
I was also thinking "bad request", but made the API more forgiving. I'll revisit this with the other comments next week.